pkg/spoke/submarineragent/config_controller.go

```diff
@@ -154,6 +152,16 @@ func (c *submarinerConfigController) sync(ctx context.Context, syncCtx factory.S
 			return nil
 		}
 
+		if config.Status.ManagedClusterInfo.Platform == "GCP" {
+			cloudProvider, preparedErr := cloud.GetCloudProvider(c.restMapper, c.kubeClient, nil, c.dynamicClient,
+				c.hubKubeClient, c.eventRecorder, config.Status.ManagedClusterInfo, config)
+			if preparedErr == nil {
+				preparedErr = cloudProvider.CleanUpSubmarinerClusterEnv()
+			}
+
+			return nil
```

Comment:
Returning here will skip removing the gateways below. I think we only want to return on error.

```suggestion
			if preparedErr != nil {
			    return preparedErr
			{
```

Reply:
done.